### PR TITLE
[feat] render podman container/pod ports as ranges

### LINKED ISSUE #1378: Show pod/container ports as a range when possible
When there are multiple ports all in the same range, podman shows them as `50499-50504` instead of per port as this can get rather cluttered
image

image

### PR DESCRIPTION
**Pull request by benniekiss**
This PR allows rendering podman container ports  as  ranges.

It makes sure to only render a range if the range of ports has the same protocol and IP and the host ports are an equivalent range.

This is my first PR in javascript, so please let me know if anything needs fixed!

Closes #1378 

Examples:
Screenshot_20240829_135103

### CODE REVIEW

**Review by jelly — COMMENTED** [review]
Thanks for the PR, I'm happy to review it. Great to see you added tests, they are currently failing:

```
testlib.Error: timeout
wait_js_cond(ph_in_text("#containers-containers tr:contains(\"localhost/test-busybox\") dt:contains(\"Ports\") + dd","0.0.0.0:4001-4004 \u2192 4001-4004/tcp")): Error: actual text: 0.0.0.0:4000-4003 → 4000-4003/tcp0.0.0.0:6000 → 5000/tcp127.0.0.1:6001 → 5001/udp0.0.0.0:44965 → 8001/tcp127.0.0.2:37797 → 9001/tcp
```

Something went wrong in the text encoding.

File: test/check-application [hunk 1]
```diff
@@ -2035,9 +2048,12 @@ class TestApplication(testlib.MachineCase):
                        ' \u2192 8001/tcp')
         b.wait_not_in_text(f'#containers-containers tr:contains("{IMG_BUSYBOX}") dt:contains("Ports") + dd',
                            '7001/tcp')
+        b.wait_in_text(f'#containers-containers tr:contains("{IMG_BUSYBOX}") dt:contains("Ports") + dd',
+                       '0.0.0.0:4001-4004 \u2192 4001-4004/tcp')
```
Comment (jelly) [inline test/check-application:2052]: `\u2192` should be `→`

Comment (benniekiss) [reply test/check-application:2052]: I'm not actually sure what's going wrong here. I mirrored the tests above this one, and they use the same escape code/syntax

Comment (jelly) [reply test/check-application:2052]: I've build a VM locally, I'll test and help you figure it out

Comment (jelly) [reply test/check-application:2052]: image

So there is no 4001-4004 range here

File: src/ContainerIntegration.jsx [hunk 2]
```diff
@@ -15,17 +15,57 @@ export const renderContainerPublishedPorts = ports => {
...
+            let isHostPortConsecutive = false
+            let isSameHostIP = false
+            let isSameProtocol = false
+
+            if (lastRange) {
+                isPortConsecutive = portNumber === lastRange.endPort + 1;
+                isHostPortConsecutive = hostPort === lastRange.hostEndPort + 1;
+                isSameHostIP = hostIP === lastRange.hostIp;
+                isSameProtocol = proto === lastRange.protocol;
+            }
```
Comment (jelly) [inline src/ContainerIntegration.jsx:41]: minor nitpick, I'd prefer a newline between this and the following if statement.

File: test/check-application (hunk 1 above)
Comment (benniekiss) [reply test/check-application:2052]: Oh my gosh, thank you! It's clear now in the first log message you sent 😅

**Comment by jelly** [comment]
CentOS 10 is some RC/Beta issue, triggering the integration tests

**Comment by jelly** [comment]
```

# /work/make-checkout-workdir/src/ContainerIntegration.jsx
#   31:42  error  Missing semicolon  semi
#   32:46  error  Missing semicolon  semi
#   33:37  error  Missing semicolon  semi
#   34:39  error  Missing semicolon  semi
```

eslint fails on missing semicolons, can you add them? Thanks!

File: src/ContainerIntegration.jsx [hunk 3]
```diff
@@ -15,17 +15,58 @@ export const renderContainerPublishedPorts = ports => {
...
-        (hostBindings ?? []).forEach(binding => { // not-covered: null was observed in the wild, but unknown how to reproduce
-            items.push(
-                <ListItem key={ containerPort + binding.HostIp + binding.HostPort }>
-                    { binding.HostIp || "0.0.0.0" }:{ binding.HostPort } &rarr; { containerPort }
-                </ListItem>
-            );
+        const [port, proto] = containerPort.split('/');
+        const portNumber = Number(port);
+        // not-covered: null was observed in the wild, but unknown how to reproduce
+        (hostBindings ?? []).forEach(binding => {
```
Comment (cockpituous) [inline src/ContainerIntegration.jsx:26]: This added line is not executed by any test.

File: src/ContainerIntegration.jsx [hunk 4]
```diff
@@ -15,17 +15,58 @@ export const renderContainerPublishedPorts = ports => {
...
+            const lastRange = ranges[ranges.length - 1];
+            const hostIP = binding.HostIp || "0.0.0.0";
```
Comment (cockpituous) [inline src/ContainerIntegration.jsx:28]: This added line is not executed by any test.

**Comment by jelly** [comment]
I've taken the liberty to rebase and squash your changes into one commit as I would like to land it today.

**Comment by benniekiss** [comment]
Great! Thanks for reviewing this and working with me

File: src/ContainerIntegration.jsx (hunk 3 above)
Comment (cockpituous) [inline src/ContainerIntegration.jsx:26]: This added line is not executed by any test.

File: src/ContainerIntegration.jsx (hunk 4 above)
Comment (cockpituous) [inline src/ContainerIntegration.jsx:28]: This added line is not executed by any test.

**Review by jelly — APPROVED** [review]
Thanks!

**Comment by jelly** [comment]
rpm-build:centos-stream-10-x86_64 — RPMs failed to be built. 

DNF issues outside of us.

**Pull request merged**